File: 2230.java
//* @author:
import java.util.*;

class Edge{
    int v, next;
    boolean vis;
    Edge(int v,int next){
        this.v = v;
        this.next = next;
        this.vis = false;
    }
}
class Graph{
    static int MAXN = 10010;
    static int n, m, size;
    static int[] ans = new int[MAXN*3];
    static int[] head = new int[MAXN];
    static Edge[] edge = new Edge[MAXN*10];
    Graph(){
        Arrays.fill(head,-1);
        size = 0;
    }
    void add_edge(int u,int v){
        edge[size] = new Edge(v,head[u]);
        head[u] = size ++;
    }

    void Find_Euler(int u){
        for(int i = head[u];i != -1;i = edge[i].next){
            if(!edge[i].vis){
                edge[i].vis = true;
                Find_Euler(edge[i].v);
            }
        }
        System.out.println(u);
    }

}
public class Main{
    public static void main(String []args){
        Scanner in = new Scanner(System.in);
        while(in.hasNext()){
            Graph a = new Graph();
            a.n = in.nextInt();
            a.m = in.nextInt();
            for(int i = 1;i <= a.m; ++i){
                int u = in.nextInt();
                int v = in.nextInt();
                a.add_edge(u,v);
                a.add_edge(v,u);
            }
            a.Find_Euler(1);
        }
    }
}

File: 2230.cc
#include <iostream>
#include <cstdlib>
#include <cstring>
#include <cstdio>
using namespace std;

#define maxn 10005
#define maxm 100005

struct Edge {
  int v, next;
} edge[maxm];

int n, m;
int head[maxn];
int ecount;
bool vis[maxm];

void addedge(int a, int b) {
  edge[ecount].v = b;
  edge[ecount].next = head[a];
  head[a] = ecount++;
}

void dfs(int a) {
  for (int i = head[a]; i != -1; i = edge[i].next) {
    if (vis[i]) continue;
    int v = edge[i].v;
    vis[i] = true;
    dfs(v);
  }
  printf("%d\n", a + 1);
}

int main() {
  memset(head, -1, sizeof(head));
  ecount = 0;
  scanf("%d%d", &n, &m);
  for (int i = 0; i < m; i++) {
    int a, b;
    scanf("%d%d", &a, &b);
    a--;
    b--;
    addedge(a, b);
    addedge(b, a);
  }
  memset(vis, 0, sizeof(vis));
  dfs(0);
  return 0;
}
